code/modules/events/crystal_event.dm

```diff
@@ -48,9 +47,100 @@ This section is for the event controller
+			else
+				turf_loc.ChangeTurf(/turf/closed/indestructible/crystal_wall)
+
+	var/list/crystal_spawner_turfs = list()
+	for(var/turf/range_turf in RANGE_TURFS(6, dest_crystal.loc))
+		if(!isopenturf(range_turf) || isspaceturf(range_turf))
+			continue
+		crystal_spawner_turfs += range_turf
+
+	for(var/i in 1 to 6)
+		var/pick_portal = pickweight(GLOB.crystal_invasion_waves["big wave"])
+		var/turf/portal_spawner_turf = pick(crystal_spawner_turfs)
+		new pick_portal(portal_spawner_turf)
+
+///Chooses random areas where to put the main huge portals and then spawn around them 6 more portals
+/datum/round_event/crystal_invasion/proc/create_random_portals()
+	var/list/center_finder = list()
+	for(var/es in GLOB.generic_event_spawns)
+		var/obj/effect/landmark/event_spawn/temp = es
+		if(is_station_level(temp.z))
+			center_finder += temp
+	if(!center_finder.len)
+		kill()
+		CRASH("No landmarks on the station map, aborting")
+	for(var/j in 1 to portal_numbers)
+		var/obj/center = pick(center_finder)
+		var/turf/center_turf = center.loc
+		var/area/center_area = get_area(center_turf)
+		center_areas += center_area
+
+		explosion(center_turf,0,0,5,7,7)
+
+		new/obj/structure/crystal_portal/huge(center_turf)
+
+		for(var/turf/turf_loc in RANGE_TURFS(5, center_turf))
+			var/distance_from_center = max(get_dist(turf_loc, center_turf), 1)
+			if(prob(250 / distance_from_center))
+				if(isopenturf(turf_loc) || isspaceturf(turf_loc))
+					turf_loc.ChangeTurf(/turf/open/indestructible/crystal_floor, flags = CHANGETURF_INHERIT_AIR)
+				else
+					turf_loc.ChangeTurf(/turf/closed/indestructible/crystal_wall)
+
+		var/list/portal_spawner_turfs = list()
+		for(var/turf/range_turf in RANGE_TURFS(6, center_turf))
+			if(!isopenturf(range_turf) || isspaceturf(range_turf))
+				continue
+			portal_spawner_turfs += range_turf
+
+		for(var/i in 1 to 6)
+			if(!length(portal_spawner_turfs))
+				break
+			var/pick_portal = pickweight(GLOB.crystal_invasion_waves[wave_name])
+			var/turf/portal_spawner_turf = pick_n_take(portal_spawner_turfs)
+			new pick_portal(portal_spawner_turf)
+	dest_crystal.icon_state = "psy_shielded"
+	dest_crystal.changed_icon = FALSE
+	addtimer(CALLBACK(src, .proc/announce_locations), 8 SECONDS)
+
+///After 8 seconds from the initial explosions centcom will announce the location of the huge portals
+/datum/round_event/crystal_invasion/proc/announce_locations()
+	priority_announce("WARNING - After tracking the powerspikes from your station we have determined that huge portals have appeared at the following locations:[center_areas.Join(", ")]. \
+						Please close those before attempting to stabilize the crystal.", "Alert")
+	sound_to_playing_players('sound/misc/notice1.ogg')
 
 /datum/round_event/crystal_invasion/announce(fake)
 	priority_announce("WARNING - Destabilization of the Supermatter Crystal Matrix detected, please stand by waiting further instructions", "Alert")
```

Comment:
"please stand by and await further instructions." would sound more natural to me